### PR TITLE
Prefs [TASK][27913] Create splash screen

### PR DESCRIPTION
**Pull request by NguyenTaTrung**
## Related Tickets
- [#Task 27913](https://edu-redmine.sun-asterisk.vn/issues/27913)

## WHAT
- Create splash screen
## HOW
- Design logo app
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Reviewer1 (TrungNT) | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
General |Code should be self-explanatory. | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General |Is this function or class too big? If yes, is the function or class having too many responsibilities? | | NO |<li>- [x] no</li>|<li>- [ ] no</li>  
General | Does the code follow DRY?: Do not Repeat Yourself | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General |Exception handling and cleanup (dispose) resources | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General | Does the code build and execute correctly? | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Conventions | Does the source code have end-line errors? | | NO |<li>- [x] no</li>|<li>- [ ] no</li>  
Conventions | Do the source code have end-line errors? | | NO |<li>- [x] no</li>|<li>- [ ] no</li> 
Conventions | No hard coding, use constants/configuration values | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/sun/trainingcourse/anticovid_19/ui/splash/SplashActivity.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import sun.trainingcourse.anticovid_19.ui.main.MainActivity
+
+class SplashActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        startActivity(Intent(this, MainActivity::class.java))
```
Comment (huytq-2422) [inline app/src/main/java/sun/trainingcourse/anticovid_19/ui/splash/SplashActivity.kt]: Conventions 2.11 
```suggestion
        startActivity(MainActivity.getIntent(this))
```

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -9,13 +9,15 @@
...
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
 
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <activity android:name=".ui.main.MainActivity" />
     </application>
 
 </manifest>
```
Comment (aug16vcc) [inline app/src/main/AndroidManifest.xml]: Thêm dòng trống cuối file hoặc tìm setting `ensure line feed at file end on Save` rồi bật lên nhé

**Review by aug16vcc — APPROVED** [review]

File: app/src/main/java/sun/trainingcourse/anticovid_19/base/BaseFragment.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import java.lang.RuntimeException
+
+
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/sun/trainingcourse/anticovid_19/base/BaseFragment.kt:10]: blank

File: app/src/main/java/sun/trainingcourse/anticovid_19/base/BaseFragment.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        super.onCreateView(inflater, container, savedInstanceState)
+        return inflater.inflate(layoutResource, container, false)
+    }
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        try {
+            fragmentInteractionCallback = context as FragmentInteractionCallback
+        } catch (e: ClassCastException) {
+            throw RuntimeException(context.toString() + " " + FragmentInteractionCallback::class.java.name)
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/sun/trainingcourse/anticovid_19/base/BaseFragment.kt]: nên viết ra 1 phương thức trả về String thay vì như thế này

Comment (NguyenTaTrung) [reply app/src/main/java/sun/trainingcourse/anticovid_19/base/BaseFragment.kt]: Dạ vâng a

**Review by tuannm-1276 — APPROVED** [review]

**Pull request merged**